### ISSUE TITLE
All HA entities transiently revert to a specific value when HA restarts 

### ISSUE THREAD

**Issue by hhinma** [bachya/ecowitt2mqtt#349]
**Describe the bug**

HA version 2022.10 (but problem existed on multiple previous versions.
ecowitt2mqtt built from latest on 2022-10-11

When HA restarts, all temperature, heat index, and dew point humidity entities revert briefly to some (previous?) transient value. It is not the same value each time, and so far, I have been unable to find any kind of pattern. This behavior is entirely repeatable and happens every time HA is restarted.  This behavior has existed for many months over multiple versions of HA - only now getting to address it.   It is unclear if this is a HA problem or an ecowitt2mqtt problem, although I am leaning towards some oddity of how the discovery topic gets processed by HA.  This entire problem is suggestive of some kind of mismatch between HA and how the topics are published.

The attached file shows the behavior over the last week, each of the transients visible in this graphic corresponds to a HA restart (ignore the barometric pressure - that's from a completely different sensor)

Essentially,  following HA restart,  sensors show "unavailable" as expected until the first update occurs, at which point the transient appears.  Following that, the correct value appears.    I have multiple MQTT-based sensors of different types, and non of them behave in this way other than the ecowitt2mqtt ones.

Has this been seen before?

**To Reproduce**
Simply restart HA with active ecowitt2mqtt with WH31 and WH32 sensors.   Transient value appears recorded in the database. 

**Diagnostics**
Output from using the `--diagnostics` flag.

**Additional context**
![ecowitt-transients](https://user-images.githubusercontent.com/8053813/196460420-4a9cb6cc-9e0c-4909-9bc8-0fcc7cfbc85a.jpg)

**Comment by bachya** [comment]
Do you have `--mqtt-retain` set?

**Comment by hhinma** [comment]
Not that I am aware of - my initial thought was that it was related to retained pubs because of the odd "switching back to some previous value",  but I do not see any evidence of that in looking at what is in the broker. 

This is run with a trivial docker-compose file that looks like the following

`version: "3"

services:

  ecowitt2mqtt:
    build: .
    container_name: ecowitt2mqtt
    environment:
      ECOWITT2MQTT_VERBOSE: "false"
      ECOWITT2MQTT_ENDPOINT: /data/report
      ECOWITT2MQTT_HASS_DISCOVERY: "true"
      ECOWITT2MQTT_HASS_DISCOVERY_PREFIX: "homeassistant"
      ECOWITT2MQTT_PORT: 8090
      ECOWITT2MQTT_BROKER: 10.2.1.201
      MQTT_BROKER: 10.2.1.201
      ECOWITT2MQTT_MQTT_PORT: 1883
      ECOWITT2MQTT_MQTT_PASSWORD: xxxxxx
      ECOWITT2MQTT_MQTT_USERNAME: xxxxxx
    image: "bachya/ecowitt2mqtt:latest"
    ports:
      - "8090:8090/tcp"
    restart: always`

**Comment by bachya** [comment]
I'm guessing that when HASS restarts and requests data from the broker, it doesn't exist because it isn't being retained. Try setting `ECOWITT2MQTT_MQTT_RETAIN` to `"true"`, restart ecowitt2mqtt, and see if the problem resolves.

**Comment by hhinma** [comment]
Yup that solved the problem.  Thank you for pointing out the obvious thing I missed.  I supposed I got intellectually sidetracked into trying to devise a mechanism to cause the specific values I was seeing instead of attacking the problem directly.  Looking back at the suite of air quality sensors that I implemented a few years back,  I noticed that the python server I coded to drive them uses retained pubs - which is why they did not suffer from the same problem.

I am still curious as to where the actual values that appeared came from - and the fact that they were different each time.  Uninitialized data somewhere in HA, I would imagine.
